File: app/src/main/java/com/mat/app3/transference/ReportTabbetActivity.java
package com.mat.app3.transference;

import android.content.Context;
import android.content.Intent;
import android.graphics.Rect;
import android.net.Uri;
import android.os.Bundle;
import android.util.Log;
import android.view.MotionEvent;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.tabs.TabLayout;

import com.mat.app3.R;
import com.mat.app3.database.entities.Report;
import com.mat.app3.models.ReportTestModel;
import com.mat.app3.transference.ui.main.ReporBasicDataFragment;
import com.mat.app3.transference.ui.main.ResportJobsFragment;
import com.mat.app3.transference.ui.main.SectionsPagerAdapter;

import java.util.ArrayList;

import java.util.Objects;

public class ReportTabbetActivity extends AppCompatActivity
        implements
        ReporBasicDataFragment.OnFragmentInteractionListener,
        ResportJobsFragment.OnFragmentInteractionListener {

    private static FloatingActionButton fab;
    private static ViewPager viewPager;
    private static TabLayout tabs;
    private static SectionsPagerAdapter sectionsPagerAdapter;
    ArrayList<String> tittles = new ArrayList<>();

    private static String TAG = ReportTabbetActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_tabbet);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        Objects.requireNonNull(getSupportActionBar()).setDisplayHomeAsUpEnabled(true);

        Bundle b = getIntent().getExtras();



        declare();
        events();
    }

    private void events() {
        fab.setOnClickListener(v->{


        });
    }

    private void declare() {
        tittles = new ArrayList<>();
        tittles.add(getString(R.string.tab_text_1));
        tittles.add(getString(R.string.tab_text_2));
        PageViewModel.init();
        ReportTestModel reportTestModel = new ReportTestModel();
        PageViewModel.set(reportTestModel);
        sectionsPagerAdapter = new SectionsPagerAdapter( getSupportFragmentManager(),tittles);

        fab = findViewById(R.id.fab);
        viewPager = findViewById(R.id.view_pager);
        viewPager.setAdapter(sectionsPagerAdapter);

        tabs = findViewById(R.id.tabs);
        tabs.setupWithViewPager(viewPager);

    }


    @Override
    public boolean dispatchTouchEvent(MotionEvent event) {
        if (event.getAction() == MotionEvent.ACTION_DOWN) {
            View v = getCurrentFocus();
            if ( v instanceof EditText) {
                Rect outRect = new Rect();
                v.getGlobalVisibleRect(outRect);
                if (!outRect.contains((int)event.getRawX(), (int)event.getRawY())) {
                    v.clearFocus();
                    InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
                    assert imm != null;
                    imm.hideSoftInputFromWindow(v.getWindowToken(), 0);
                }
            }
        }
        return super.dispatchTouchEvent( event );
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }

    @Override
    public void onFragmentInteraction_upd_eTextDNI(String mensaje) {

    }

    @Override
    public void onFragmentInteraction(Uri uri) {

    }


}

File: app/src/main/java/com/mat/app3/Report/ReportTabbetActivity.kt
package com.mat.app3.Report

import android.content.Context
import android.graphics.Rect
import android.net.Uri
import android.os.Bundle
import android.view.MotionEvent
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.EditText
import androidx.appcompat.app.ActionBar

import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.viewpager.widget.ViewPager

import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.google.android.material.tabs.TabLayout

import com.mat.app3.R
import com.mat.app3.database.MyDatabase
import com.mat.app3.Report.ui.main.ReporBasicDataFragment
import com.mat.app3.Report.ui.main.ResportJobsFragment
import com.mat.app3.Report.ui.main.SectionsPagerAdapter

import java.util.ArrayList

import java.util.Objects

class ReportTabbetActivity : AppCompatActivity(),
    ReporBasicDataFragment.OnFragmentInteractionListener,
    ResportJobsFragment.OnFragmentInteractionListener {
    internal var tittles = ArrayList<String>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_tabbet)

        val toolbar = findViewById<View>(R.id.toolbar) as Toolbar
        setSupportActionBar(toolbar)
        Objects.requireNonNull<ActionBar>(supportActionBar).setDisplayHomeAsUpEnabled(true)

        val b = intent.extras



        declare()
        events()
    }



    private fun events() {
        fab!!.setOnClickListener { v ->


        }
    }

    private fun declare() {
        tittles = ArrayList()
        tittles.add(getString(R.string.tab_text_1))
        tittles.add(getString(R.string.tab_text_2))

        Thread{
            PageViewModel.init()
            PageViewModel.setPost(parseToListJobs.parse(MyDatabase.getInstance(this).reportDao().selectAll().get(0)))
        }.start()

        sectionsPagerAdapter = SectionsPagerAdapter(supportFragmentManager, tittles)

        fab = findViewById(R.id.fab)
        viewPager = findViewById(R.id.view_pager)
        viewPager!!.adapter = sectionsPagerAdapter

        tabs = findViewById(R.id.tabs)
        tabs!!.setupWithViewPager(viewPager)

    }


    override fun dispatchTouchEvent(event: MotionEvent): Boolean {
        if (event.action == MotionEvent.ACTION_DOWN) {
            val v = currentFocus
            if (v is EditText) {
                val outRect = Rect()
                v.getGlobalVisibleRect(outRect)
                if (!outRect.contains(event.rawX.toInt(), event.rawY.toInt())) {
                    v.clearFocus()
                    val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
                    imm.hideSoftInputFromWindow(v.windowToken, 0)
                }
            }
        }
        return super.dispatchTouchEvent(event)
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }

    override fun onFragmentInteraction_upd_eTextDNI(mensaje: String) {

    }

    override fun onFragmentInteraction(uri: Uri) {

    }

    companion object {

        private var fab: FloatingActionButton? = null
        private var viewPager: ViewPager? = null
        private var tabs: TabLayout? = null
        private var sectionsPagerAdapter: SectionsPagerAdapter? = null

        private val TAG = ReportTabbetActivity::class.java.simpleName
    }


}